loadtest/control/gencontroller/actions.go

```diff
@@ -422,7 +415,7 @@ func (c *GenController) joinChannel(u user.User) control.UserActionResponse {
 		}
 		resp = control.UserActionResponse{Info: fmt.Sprintf("joined channel %s", channelID)}
 
-		if err := c.user.GetPostsForChannel(channelID, 0, 60, collapsedThreads); err != nil {
+		if err := c.sysadmin.GetPostsForChannel(channelID, 0, 60, collapsedThreads); err != nil {
```

Comment:
Slightly unrelated question but how are we guaranteeing that users will have some posts in store to pick from, for example, to reply or react?

Reply:
Aha good point there.. yeah that's a bug. I used sysadmin because, normal users weren't able to load posts from private channels. Which is why I used sysadmin for `AddChannelMember`  as well. Any suggestions?

Reply:
The problem is that after https://github.com/mattermost/mattermost-load-test-ng/pull/604 we essentially removed the call except for this case so it may become an issue even if we fix just this one. As things stand I suppose users will only reply and react to their own posts. 

I think we need to find a compromise to at least load posts every now and then, not as often as before to avoid killing the db but with some lower frequency maybe through a dedicated action. Another option, although likely more complex, would be to hook up the websocket to get the broadcasts but it may be overkill.

Reply:
I am currently writing a solution where I just use the sysadmin HTTP client to make the API call, and then use the normal user's store to store the posts. Essentially moving up the userentity logic to this upper layer. It's not nice, but then we need to do the same for `AddChannelMember` as well. Because otherwise, the channel members get added to the sysadmin's store, not the current user's store.